scipy/stats/_hypotests.py

```diff
@@ -1990,38 +1986,30 @@ def tukey_hsd(*args):
     ntreatments = len(args)
     means = np.asarray([np.mean(arg) for arg in args])
     nsamples_treatments = np.asarray([a.size for a in args])
-    nobs = np.sum(nsamples_treatments)
-
-    # determine mean square error [5]. Note that this is sometimes called
-    # mean square error within.
-    mse = (np.sum([np.var(arg, ddof=1) for arg in args] *
-                  (nsamples_treatments - 1)) / (nobs - ntreatments))
-
-    # The calculation of the standard error differs when treatments differ in
-    # size. See ("Unequal sample sizes")[1].
-    if np.unique(nsamples_treatments).size == 1:
-        # all input groups are the same length, so only one value needs to be
-        # calculated [1].
-        normalize = 2 / nsamples_treatments[0]
-    else:
-        # to compare groups of differing sizes, we must compute a variance
-        # value for each individual comparison. Use broadcasting to get the
-        # resulting matrix. [3], verified against [4] (page 308).
-        normalize = 1 / nsamples_treatments + 1 / nsamples_treatments[None].T
-
-    # the standard error is used in the computation of the tukey criterion and
-    # finding the p-values.
-    stand_err = np.sqrt(normalize * mse / 2)
 
-    # the mean difference is the test statistic.
+    vars_ = np.asarray([np.var(arg, ddof=1) for arg in args])
     mean_differences = means[None].T - means
 
-    # Calculate the t-statistic to use within the survival function of the
-    # studentized range to get the p-value.
-    t_stat = np.abs(mean_differences) / stand_err
+    # the denominator Behrens-Fisher statistic ref.7 p.5 with notation $v$
+    # calculated by sample variance and sample size
+    # the squre root of 2 is considered to calculate the stand_err
+    stand_err = np.sqrt(
+        ((vars_ / nsamples_treatments)[None].T
+         + (vars_ / nsamples_treatments)) / 2)
+
+    # Welch degree of freedom ref.7 p.5 with notation $\nu$
+    nus = (
+            ((vars_/nsamples_treatments)[None].T + (vars_/nsamples_treatments))**2
+            / ((((vars_/nsamples_treatments)**2)/(nsamples_treatments-1))[None].T
+            + (((vars_/nsamples_treatments)**2)/(nsamples_treatments-1)))
+    )
+
+    t_stat = mean_differences / stand_err
 
-    params = t_stat, ntreatments, nobs - ntreatments
+    # H0 was rejected if $|v| \ge q(\alpha, K, \nu)/\sqrt{2}$
+    # ref.7 p.6
+    params = abs(t_stat), ntreatments, nus
     pvalues = distributions.studentized_range.sf(*params)
 
-    return TukeyHSDResult(mean_differences, pvalues, ntreatments,
-                          nobs, stand_err)
+    return GamesHowellResult(mean_differences, pvalues, ntreatments,
```

Comment:
Let's just keep calling this `TukeyHSDResult`. It is a result object from the `tukey_hsd` function.